adafruit_ads7830/ads7830.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 Liz Clark for Adafruit Industries
+#
+# SPDX-License-Identifier: MIT
+"""
+:py:class:`~adafruit_ads7830.ads7830.ADS7830`
+================================================================================
+
+CircuitPython driver for the ADS7830 analog to digital converter
+
+
+* Author(s): Liz Clark
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit ADS7830 8-Channel 8-Bit ADC with I2C <https://www.adafruit.com/product/5836>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://circuitpython.org/downloads
+
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+"""
+
+from adafruit_bus_device.i2c_device import I2CDevice
+from micropython import const
+
+try:
+    import typing  # pylint: disable=unused-import
+    from busio import I2C
+except ImportError:
+    pass
+
+__version__ = "0.0.0+auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_ADS7830.git"
+
+_I2C_ADDR = const(0x48)
+
+
+class ADS7830:
+    """Adafruit ADS7830 ADC driver"""
+
+    _POWER_DOWN_MODES = [
+        (True, True),  # power down ref and adc
+        (True, False),  # power down ref and not adc
+        (False, True),  # power down adc and not ref
+        (False, False),  # do not power down ref or adc
+    ]
+    # Single channel selection list
+    _CHANNEL_SELECTION = [
+        0x08,  # SINGLE_CH0
+        0x0C,  # SINGLE_CH1
+        0x09,  # SINGLE_CH2
+        0x0D,  # SINGLE_CH3
+        0x0A,  # SINGLE_CH4
+        0x0E,  # SINGLE_CH5
+        0x0B,  # SINGLE_CH6
+        0x0F,  # SINGLE_CH7
+    ]
+    # Differential channel selection list
+    _DIFF_CHANNEL_SELECTION = [
+        0x00,  # DIFF_CH0_CH1
+        0x04,  # DIFF_CH1_CH0
+        0x01,  # DIFF_CH2_CH3
+        0x05,  # DIFF_CH3_CH2
+        0x02,  # DIFF_CH4_CH5
+        0x06,  # DIFF_CH5_CH4
+        0x03,  # DIFF_CH6_CH7
+        0x07,  # DIFF_CH7_CH6
+    ]
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        i2c: I2C,
+        address: int = _I2C_ADDR,
+        diff_mode: bool = False,
+        int_ref_pd: bool = False,
+        adc_pd: bool = False,
+    ) -> None:
+        """Initialization over I2C
+
+        :param int address: I2C address (default 0x48)
+        :param bool diff_mode: Select differential vs. single mode
+        :param bool int_ref_pd: Power down mode for internal reference (defaults to False)
+        :param bool adc_pd: Power down mode for ADC (defaults to False)
+        """
+        self.i2c_device = I2CDevice(i2c, address)
+        _pd = (int_ref_pd, adc_pd)
+        self.power_down = self._POWER_DOWN_MODES.index(_pd)
```

Comment:
```suggestion
        pd = 0
        if not int_ref_pd:
            pd |= 2
        if not adc_pd:
            pd |= 1
        self.power_down = pd
```

And remove the list above. It isn't needed.

Reply:
oh cool, didn't realize that for the power down list. got rid of the list and updated some of the names to try and make them more clear